kasa/smart/modules/energy.py

```diff
@@ -105,3 +109,11 @@ async def get_daily_stats(self, *, year=None, month=None, kwh=True) -> dict:
     async def get_monthly_stats(self, *, year=None, kwh=True) -> dict:
         """Return monthly stats for the given year."""
         raise KasaException("Device does not support periodic statistics")
+
+    async def _check_supported(self):
+        """Additional check to see if the module is supported by the device."""
+        # Energy module is not supported on P304M parent device
+        return not (
+            self._device.model == "P304M"
```

Comment:
I hope we can find a way to avoid hardcoding any models information. If there's no `_check_supported` at all, does it still work (albeit with an error  code for the query)?

Reply:
Completely understand not wanting to use the model here, this was the "least bad" option I could come up with.

If I remove the `_check_supported` implementation, the result is a warning with an error 1001 from the parent (generic error code?) but the child devices do correctly report their energy usage.

Reply:
So we have this issue with the KS240 as well which reports the child components at the parent level. The solution after many iterations was to look for a related key in the device_info as part of the `_check_supported`. The `Brightness` and `Fan` modules are examples of this. I'd suggest the `device_on` could be a suitable key although worth checking the other fixtures to make sure that key is always present when there is a `get_energy_usage` response.

Reply:
Had a look @sdb9696 and you're absolutely correct, based on the current fixtures then `device_on` is always present on devices that include energy monitoring. I had only considered fields were directly related to energy monitoring but as of right now, TP-Link/Tapo only seem to offer that on devices that are also switches.

I've tested things locally with your suggested change and all looks good to me! Tests for all existing fixtures continue to pass and querying the P304M from the `kasa` CLI doesn't throw errors. Many thanks.

Reply:
Pushed with suggested changes.

Reply:
Awesome! 🎉 Feel free to remove the fixture and related changes from this PR, and we can merge this followed by the second fixture PR just after that.